### ISSUE TITLE
ZAP uses RetireJs to detect these vulnerabilities, but it looks like thats right: https://github.com/RetireJS/retire.js/blob/master/repository/jsrepository.json#L1485-L1508

### ISSUE THREAD

**Issue by robmro27** [zaproxy/zaproxy#7926]
ZAP uses RetireJs to detect these vulnerabilities, but it looks like thats right: https://github.com/RetireJS/retire.js/blob/master/repository/jsrepository.json#L1485-L1508
```
	"bootstrap": {
		"vulnerabilities" : [
			{
				"below" : "4.3.1",
				"atOrAbove" : "4.0.0",
				"identifiers": {
					"issue" : "28236",
					"summary": "XSS in data-template, data-content and data-title properties of tooltip/popover",
					"CVE" : ["CVE-2019-8331"]
				},
				"severity" : "medium",
				"info" : [ "https://github.com/twbs/bootstrap/issues/28236" ]
			},
			{
				"below" : "3.4.1",
				"identifiers": {
					"issue" : "28236",
					"summary": "XSS in data-template, data-content and data-title properties of tooltip/popover",
					"CVE" : ["CVE-2019-8331"]
				},
				"severity" : "medium",
				"info" : [ "https://github.com/twbs/bootstrap/issues/28236" ]
			},
```
Could be a bug in our version calculating code?

_Originally posted by @psiinon in https://github.com/zaproxy/zaproxy/issues/6818#issuecomment-923738423_

**Comment by thc202** [comment]
Better to follow the issue template, what FP are you seeing?

**Comment by thc202** [comment]
Could you provide the CVE as well?

**Comment by robmro27** [comment]
I belive it's in old task https://www.cvedetails.com/cve/CVE-2019-8331/

**Comment by thc202** [comment]
It does not seem to be the same issue, the version reported is `3.4.1` while in the other it was reported as `3.4.1.min`.

**Comment by thc202** [comment]
Seems that's caused by outdated info, we are still using this version https://github.com/RetireJS/retire.js/blob/9c453a8d6ee0aa64578022957602da2f0beed8d0/repository/jsrepository.json#L2718

**Comment by robmro27** [comment]
But it also shows below so 3.4.1 should be clear acctually it's same as uptodate ?

**Comment by thc202** [comment]
True it should not report for 3.4.1 with `below`, can you confirm the other info has `CVE-2019-8331` ?

**Comment by robmro27** [comment]
Other info? Sorry could you explain what you wuld like me to check?

**Comment by thc202** [comment]
In the Alert tab, under the Alert details the field "Other Info".

**Comment by robmro27** [comment]
It's empty

![image](https://github.com/zaproxy/zaproxy/assets/4953153/175a31f2-afb7-45e4-869c-9c4c3812202d)

**Comment by thc202** [comment]
Thank you! That confirms it's matching the other entry.

**Comment by robmro27** [comment]
But not this you provided I think because in my case it's medium, and you show low

**Comment by thc202** [comment]
We don't use the "severity" field, the alerts are all raised with Medium risk. https://github.com/zaproxy/zap-extensions/blob/04d6b68e867b695544335477398992fa1900526b/addOns/retire/src/main/java/org/zaproxy/addon/retire/RetireScanRule.java#L99

**Comment by robmro27** [comment]
Ah ok, so I belive still there is bad info, should be informin about outdated lib but not alert it's vulnerable.

**Comment by thc202** [comment]
I guess we could check the CWE and change the details accordingly but not sure if that will have a high priority. One can check the alert references to know more about what's being reported, one could also argue that using an unmaintained library is a vulnerability waiting to happen.

**Comment by kingthorin** [comment]
> one could also argue that using an unmaintained library is a vulnerability waiting to happen.

Agreed!

**Comment by kshipp** [comment]
I agree with robmro27.  There is no basis for this to be a Medium-level alert.  It's being flagged only because it's an older version, not because it's vulnerable.  This is false reporting, IMHO.  There are no known vulnerabilities of 3.4.1, therefore, it should not be flagged as such incorrectly only because it's noted to be out of date/older version.  It should be an Informational Alert, or at best, a Low alert.

**Comment by kingthorin** [comment]
I'd be okay with us mapping to the retire.js severity field, however if it doesn't exist for a given component/entry I believe we would still default to medium.

**Comment by thc202** [comment]
https://github.com/zaproxy/zaproxy/issues/7926#issuecomment-1791466537

The hard fact is that it's no longer maintained, said that, no one is arguing that this shouldn't be changed (the issue is still open after all).
